libkbfs/folder_branch_ops.go

```diff
@@ -7424,6 +7433,173 @@ func (fbo *folderBranchOps) Reset(
 	return nil
 }
 
+func (fbo *folderBranchOps) getProtocolSyncConfig(
+	ctx context.Context, lState *lockState, kmd KeyMetadata) (
+	ret keybase1.FolderSyncConfig, err error) {
+	fbo.syncLock.AssertAnyLocked(lState)
+
+	config := fbo.config.GetTlfSyncState(fbo.id())
+	ret.Mode = config.Mode
+	if ret.Mode != keybase1.FolderSyncMode_PARTIAL {
+		return ret, nil
+	}
+
+	var block *FileBlock
+	// Skip block assembly if it's already cached.
+	b, err := fbo.config.BlockCache().Get(config.Paths.Ptr)
+	if err == nil {
+		var ok bool
+		block, ok = b.(*FileBlock)
+		if !ok {
+			return keybase1.FolderSyncConfig{}, errors.Errorf(
+				"Partial sync block is not a file block, but %T", b)
+		}
+	} else {
+		block = NewFileBlock().(*FileBlock)
+		err = assembleBlock(
+			ctx, fbo.config.keyGetter(), fbo.config.Codec(),
+			fbo.config.Crypto(), kmd, config.Paths.Ptr, block,
+			config.Paths.Buf, config.Paths.ServerHalf)
+		if err != nil {
+			return keybase1.FolderSyncConfig{}, err
+		}
+	}
+
+	paths, err := syncPathListFromBlock(fbo.config.Codec(), block)
+	if err != nil {
+		return keybase1.FolderSyncConfig{}, err
+	}
+	ret.Paths = paths.Paths
+	return ret, nil
+}
+
+// GetSyncConfig implements the KBFSOps interface for folderBranchOps.
+func (fbo *folderBranchOps) GetSyncConfig(
+	ctx context.Context, tlfID tlf.ID) (keybase1.FolderSyncConfig, error) {
+	if tlfID != fbo.id() || fbo.branch() != MasterBranch {
+		return keybase1.FolderSyncConfig{}, WrongOpsError{
+			fbo.folderBranch, FolderBranch{tlfID, MasterBranch}}
+	}
+
+	lState := makeFBOLockState()
+	md, _ := fbo.getHead(ctx, lState, mdNoCommit)
+
+	fbo.syncLock.RLock(lState)
+	defer fbo.syncLock.RUnlock(lState)
+	return fbo.getProtocolSyncConfig(ctx, lState, md)
+}
+
+func (fbo *folderBranchOps) makeEncryptedPartialPathsLocked(
+	ctx context.Context, lState *lockState, kmd KeyMetadata, paths []string) (
+	FolderSyncEncryptedPartialPaths, error) {
+	fbo.syncLock.AssertLocked(lState)
+
+	oldConfig, err := fbo.getProtocolSyncConfig(ctx, lState, kmd)
+	if err != nil {
+		return FolderSyncEncryptedPartialPaths{}, err
+	}
+	if oldConfig.Mode == keybase1.FolderSyncMode_ENABLED {
+		return FolderSyncEncryptedPartialPaths{},
+			errors.Errorf("TLF %s is already fully synced", fbo.id())
+	}
+
+	// Make sure the new path list doesn't contain duplicates,
+	// contains no duplicates, and each path is cleaned.
+	seenPaths := make(map[string]bool, len(paths))
+	var pathList syncPathList
+	pathList.Paths = make([]string, len(paths))
+	for i, p := range paths {
+		p = stdpath.Clean(filepath.ToSlash(p))
+		if seenPaths[p] {
+			return FolderSyncEncryptedPartialPaths{}, errors.Errorf(
+				"%s is in the paths list more than once", p)
+		}
+		if stdpath.IsAbs(p) {
+			return FolderSyncEncryptedPartialPaths{}, errors.Errorf(
+				"Absolute paths like %s are not allowed", p)
+		}
+		if strings.HasPrefix(p, "..") {
```

Comment:
you can make a relative path out of the TLF even if the prefix isn't `..`. You probably want `strings.Contains(p, "..")`

Reply:
@jzila I don't think you can, because of the `stdpath.Clean` call above.  It should get rid of all the unnecessary ".." components except at the beginning.